myhoard/restore_coordinator.py

```diff
@@ -1187,13 +1188,13 @@ def _start_sql_thread_if_not_running(self, slave_status):
         with self._mysql_cursor() as cursor:
             cursor.execute("START SLAVE SQL_THREAD")
 
-    def _check_sql_slave_status(self):
+    def _check_sql_slave_status(self) -> tuple[bool, int]:
         expected_range = self.state["current_executed_gtid_target"]
         expected_index = self.state["current_relay_log_target"]
 
         with self._mysql_cursor() as cursor:
-            cursor.execute("SHOW SLAVE STATUS")
-            slave_status = cursor.fetchone()
+            slave_status = get_slave_status(cursor)
+            assert slave_status
```

Comment:
I think a "real" check and exception would be a better fit than an assert here.

Auto-recovering from that odd case is out of the scope of the PR, but something like this could be more ops-friendly if that happens:
```suggestion
            if not slave_status:
                raise Exception("No status available from SHOW SLAVE STATUS, perhaps MySQL was restarted while applying binlog.")
```